pkg/daemon/daemon.go

```diff
@@ -1221,6 +1224,24 @@ func (dn *Daemon) prepUpdateFromCluster() (*mcfgv1.MachineConfig, *mcfgv1.Machin
 	return currentConfig, desiredConfig, nil
 }
 
+// getCurrentConfig tries and grab the currentConfig from the API. If, for any reason,
+// the MC isn't there, it'll go and grab it from disk.
+// see https://bugzilla.redhat.com/show_bug.cgi?id=1855821#c2
+func (dn *Daemon) getCurrentConfig(name string) (*mcfgv1.MachineConfig, error) {
+	currentConfig, err := dn.mcLister.Get(name)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			currentConfig, err = dn.getCurrentConfigOnDisk()
+			if err != nil && !os.IsNotExist(err) {
+				return nil, err
+			}
```

Comment:
Perhaps adding a log here in `else` would be useful to help us debug that something may have gone wrong due to which we are fetching the currentconfig from disk

Reply:
right, good point, I'll wrap the `err` variables to be more explicit

Reply:
Following on @sinnykumari's logging note I think it would be helpful to log that we've successfully gotten the config from disk (since [dn.getCurrentConfigOnDisk](https://github.com/openshift/machine-config-operator/blob/24e24e8e0bd2dcc79e5c429909a99d7049171bfa/pkg/daemon/daemon.go#L901-L912) doesn't log, understandably so). That way if something was, for some reason out of sync, it would be easier to trace.